anndata/tests/test_backed_sparse.py

```diff
@@ -128,13 +128,20 @@ def make_randomized_mask(size: int) -> np.ndarray:
 
 
 # non-random indices, with alternating one false and n true
-def make_alternating_mask(size: int) -> np.ndarray:
+def make_alternating_mask_10(size: int) -> np.ndarray:
     mask_alternating = np.ones(size, dtype=bool)
     for i in range(0, size, 10):  # 10 is enough to trigger new behavior
         mask_alternating[i] = False
     return mask_alternating
 
 
+def make_alternating_mask_5(size: int) -> np.ndarray:
+    mask_alternating = np.ones(size, dtype=bool)
+    for i in range(0, size, 5):  # 5 is too low to trigger new behavior
+        mask_alternating[i] = False
+    return mask_alternating
```

Comment:
```suggestion
from functools import partial

def make_alternating_mask(size: int, step: int) -> np.ndarray:
    mask_alternating = np.ones(size, dtype=bool)
    for i in range(0, size, step):  # 5 is too low to trigger new behavior
        mask_alternating[i] = False
    return mask_alternating
    
make_alternating_mask_5 = partial(make_alternating_mask, step=5)
make_alternating_mask_10 = partial(make_alternating_mask, step=10)
```

Could we do something more like this?

Reply:
Fwiw, the 5 consecutive case would really benefit from the optimization.

Reply:
This is probably laptop dependent.  I will check when I get home but I am fairly certain the cutoff point was 7.